x.py

```diff
@@ -4,6 +4,16 @@
 
 import os
 import sys
+
+# If this is python2, check if python3 is available and re-execute with that
+# interpreter.
+if sys.version_info.major < 3:
+    try:
+        os.execvp("python3", ["python3"] + sys.argv)
+    except OSError:
+        # Python 3 isn't available, do nothing
```

Comment:
Fine as-is, but "do nothing" to me sounds like the script will halt execution, whereas `pass` is a no-op

```suggestion
        # Python 3 isn't available, continue execution with Python 2
```